### ISSUE TITLE
aurutils 2.3.6b fails to upgrade a package if found on the cache

### ISSUE THREAD

**Issue by je-vv** [AladW/aurutils#741]
Hello,
See what's happening when trying to upgrade from AUR with aurutils:

```
% AUR_DEBUG=1 aur sync -st --provides -d laur -u                                                                                                    
==> Using [laur] repository
  -> python-loguru: 0.5.2-1 -> 0.5.3-1
==> Retrieving package files
fatal: destination path 'python-loguru' already exists and is not an empty directory.
```

If I remove the cache directory:  `rm -r ~/.cache/aurutils/snapshot/python-loguru`, then it works.

But even the same happened some time back during the aurutils (2.3.5-1 -> 2.3.6b-1) upgrade.  Actually then, removing the old directory was not enough, it failed later with some other issue.  I had to manually do `makepkg`, then move the generated package and its signature to the local repo, manually add it to the repo, and then do a system upgrade...

Is there something wrong with aurutils not able to handle cached package directories as before?  Is an additional setting required to actually do upgrades even when there are cached directories?

**Comment by je-vv** [comment]
Actually, here it's an example on a box trying to upgrade aurutils, and other AUR packages.  At 1st, the error:

> fatal: destination path '<package>' already exists and is not an empty directory.

Is gotten for all packages, and after removing the corresponding cache directories for those packages I get:

> ==> Using [laur] repository
>   -> aurutils: 2.3.5-1 -> 2.3.6b-1
>   -> conan: 1.29.0-1 -> 1.29.2-1
>   -> python-loguru: 0.5.2-1 -> 0.5.3-1
> ==> Retrieving package files
> Cloning into 'python-loguru'...
> remote: Enumerating objects: 61, done.
> remote: Counting objects: 100% (61/61), done.
> remote: Compressing objects: 100% (46/46), done.
> remote: Total 61 (delta 15), reused 61 (delta 15), pack-reused 0
> Unpacking objects: 100% (61/61), 8.11 KiB | 169.00 KiB/s, done.
> Cloning into 'conan'...
> remote: Enumerating objects: 686, done.
> remote: Counting objects: 100% (686/686), done.
> remote: Compressing objects: 100% (406/406), done.
> remote: Total 686 (delta 314), reused 649 (delta 279), pack-reused 0
> Receiving objects: 100% (686/686), 109.96 KiB | 280.00 KiB/s, done.
> Resolving deltas: 100% (314/314), done.
> Cloning into 'aurutils'...
> remote: Enumerating objects: 360, done.
> remote: Counting objects: 100% (360/360), done.
> remote: Compressing objects: 100% (242/242), done.
> remote: Total 360 (delta 119), reused 359 (delta 118), pack-reused 0
> Receiving objects: 100% (360/360), 69.03 KiB | 344.00 KiB/s, done.
> Resolving deltas: 100% (119/119), done.
> Running makepkg --clean --syncdeps
> ==> ERROR: build: invalid argument (no packages found)

So the option I'm left with is to remove the binary packages on the repo dir, running makepkg for each package, move the binary packages to repo dir, and then add the binary packages to the repo DB...

And afterwords, I'm left with the issue of having to remove the cached directory when upgrading...

**Comment by AladW** [comment]
2.3.5 could no longer build packages so to upgrade you'd need to run 
makepkg manually. I'm not sure on the cause of the "destination path 
already exists". Do you have similar issues with the master branch?

**Comment by AladW** [comment]
` rm -r ~/.cache/aurutils/snapshot/python-loguru` indicates there's some issue with using the `--tar` mode of `aur-fetch`. However, `aur-sync` by default uses different directories for `--tar` and `--git` checkouts. An actual traceback would solve this (2.3 does not support AUR_DEBUG unfortunately).

**Comment by je-vv** [comment]
Sorry I haven't replied before, but I was waiting for having upgrades on AUR packages twice after totally removing ~/.cache/aurutils.  See, after having the current version "2.3.6b-1" installed, I was no longer able to upgrade any package, unless I removed its respective directory in the cache 1st.  So I decided to remove the whole cache.

So far, so good, but I want to see an upgrade of a package with a new cache dir already, and see if the problem shows up or not.  It might have been there was an incompatibility as well on the caches from the prior version and the new one.

**Comment by AladW** [comment]
I wrote what you described as a test and it works for me with 3.0.0rc. https://paste.xinu.at/BBY2U/

**Comment by je-vv** [comment]
Well, I upgraded freetube, which already had a cache dir, added after I had to remove ~/.cache/aurutils, and aurutils had no issues.

So it seems that after the aurutils upgrade, it was also required to remove the whole cache generated from the prior aurutils.  Not sure if something known already...

At any rate, I no longer have issues after the aurutils upgrade, however this one was not as smooth as other ones, :).  Thanks !
